### PR TITLE
fix: Update the widget from drop down to select widget in suggested widget list

### LINKED ISSUE #13899: [Bug]: Select widget pick from Suggested Widgets is still V1
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The Select Widget V1 doesn't support theming and currently in the suggest Widget Pane, the suggested Select widget given to the user is v1

https://user-images.githubusercontent.com/91450662/168786217-ea043be9-e23b-45c5-b8fa-5936fa4b3d46.mov

### Steps To Reproduce

1. Create any new API & run to fetch results
2. Use the Select from the Suggested widgets section & add to page
3. **Observe the existing theme is not applied**
4. Drag drop another widget & see some changes the previous widget (dropdown highlight color changes)
5. **Observe : Options dropdown is overflowing outside the dropdown field**
6. Change theme, **Observe : Border Radius is still not applied & over flow also persists**
7. **Expected : Select Widget from Suggested widgets also conform to theme & its properties**

### Public Sample App

_No response_

### Version

Prod/Release v1.6.26

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description

> Update the BE code to include `Select Widget` instead of `Drop Down Widget` to support theming. 

Fixes #13899 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Locally
- JUnit Tests

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/update-widget-for-suggestion 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 56.71 **(-0.01)** | 38.66 **(-0.01)** | 36.23 **(0)** | 56.96 **(-0.01)**
 :red_circle: | app/client/src/selectors/commentsSelectors.ts | 83.61 **(-1.64)** | 61.76 **(-2.95)** | 73.33 **(0)** | 88.24 **(-2.35)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**</details>

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/WidgetSuggestionHelper.java [hunk 1]
```diff
@@ -178,9 +178,9 @@ private static List<WidgetSuggestionDTO> getWidgetsForTypeArray(List<String> fie
         List<WidgetSuggestionDTO> widgetTypeList = new ArrayList<>();
         if(!fields.isEmpty()) {
             if(fields.size() < 2) {
-                widgetTypeList.add(getWidget(WidgetType.DROP_DOWN_WIDGET, fields.get(0), fields.get(0)));
+                widgetTypeList.add(getWidget(WidgetType.SELECT_WIDGET, fields.get(0), fields.get(0)));
             } else {
-                widgetTypeList.add(getWidget(WidgetType.DROP_DOWN_WIDGET, fields.get(0), fields.get(0)));
+                widgetTypeList.add(getWidget(WidgetType.SELECT_WIDGET, fields.get(0), fields.get(0)));
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/helpers/WidgetSuggestionHelper.java:183]: Not sure what are we achieving in this `if-else` block? If not required can we please remove this.

**Review by abhvsn — COMMENTED** [review]
Minor comment on earlier implementation.

@AnaghHegde can we please confirm if we need to update `DROP_DOWN_WIDGET` with `SELECT_WIDGET` as I don't have much context here, also haven't seen any suggestion in the issue which is being resolved in this PR.

**Comment by AnaghHegde** [comment]
@abhvsn Can you please review this again ?

**Review by abhvsn — APPROVED** [review]

**Review by abhvsn — APPROVED** [review]

**Pull request merged**